internal/route/repo/view.go

```diff
@@ -120,20 +139,39 @@ func renderDirectory(c *context.Context, treeLink string) {
 	}
 }
 
-func renderFile(c *context.Context, entry *git.TreeEntry, treeLink, rawLink string) {
+func renderFile(c *context.Context, entry *git.TreeEntry, sourceEntry *git.TreeEntry, sourceTreePath string, treeLink, rawLink string) {
 	c.Data["IsViewFile"] = true
+	c.Data["IsShowRawLink"] = true
+	c.Data["RawFileLink"] = rawLink + "/" + c.Repo.TreePath
+	c.Data["EditableFileTreePath"] = c.Repo.TreePath
+
+	var blob *git.Blob
+	if entry.IsSymlink() {
+		c.Data["SourceTreePath"] = sourceTreePath
+		if sourceEntry != nil {
+			blob = sourceEntry.Blob()
+			c.Data["SourceFileSize"] = sourceEntry.Blob().Size()
+			c.Data["RawFileLink"] = rawLink + "/" + sourceTreePath
+			c.Data["EditableFileTreePath"] = sourceTreePath
+		} else {
+			c.Data["SourceFileSize"] = int64(0)
+			c.Data["IsShowRawLink"] = false
+		}
+	}
+
+	if blob == nil {
+		blob = entry.Blob()
+	}
 
-	blob := entry.Blob()
 	p, err := blob.Bytes()
 	if err != nil {
 		c.Error(err, "read blob")
 		return
 	}
 
-	c.Data["FileSize"] = blob.Size()
-	c.Data["FileName"] = blob.Name()
+	c.Data["FileSize"] = entry.Blob().Size()
+	c.Data["FileName"] = entry.Blob().Name()
```

Comment:
Why do we need to change these lines?

Reply:
Because, if the current file is a symlink, the contents of the source file should be displayed, the blob is modified as the source file.

I think that symlink have no blobs and should be source files.

So:
- `entry.Blob()` is the current file
- `blob` could be a blob of the source file

Reply:
I see. I think this is confusing, we should use `blob` and `sourceBlob`, as we do for `entry` and `sourceEntry`.